src/test/assets_tests.cpp

```diff
@@ -200,900 +204,459 @@ CreateMintAssetTx(const CTxMemPool &mempool, SimpleUTXOMap &utxos, const CKey &c
+    SignTransaction(*m_node.mempool, tx2, coinbaseKey);
+
+    {
+        auto block = std::make_shared<CBlock>(CreateBlock({tx2}, coinbaseKey));
+        EnsureChainman(m_node).ProcessNewBlock(Params(), block, true, nullptr);
+
+        BOOST_ASSERT(::ChainActive().Height() == nHeight + 3);
+        BOOST_ASSERT(block->GetHash() == ::ChainActive().Tip()->GetBlockHash());
+    }
 }
 
-assetId = tx.GetHash().ToString();
-tx = CreateMintAssetTx(*m_node.mempool, utxos, coinbaseKey, assetId);
-
+BOOST_FIXTURE_TEST_CASE(assets_invalid_cases, TestChainDIP3BeforeActivationSetup)
 {
-auto block = std::make_shared<CBlock>(CreateBlock({tx}, coinbaseKey));
-EnsureChainman(m_node)
-.
+    CKey sporkKey;
+    sporkKey.MakeNewKey(false);
+    sporkManager.SetSporkAddress(EncodeDestination(sporkKey.GetPubKey().GetID()));
+    sporkManager.SetPrivKey(EncodeSecret(sporkKey));
+    sporkManager.UpdateSpork(SPORK_22_SPECIAL_TX_FEE, 2560, *m_node.connman);
 
-ProcessNewBlock(Params(), block,
+    auto utxos = BuildSimpleUtxoMap(m_coinbase_txns);
 
-true, nullptr);
+    //create a asset
+    auto tx = CreateNewAssetTx(*m_node.mempool, utxos, coinbaseKey, "Test Asset", false, false, 0, 2, 100);
+    std::vector<CMutableTransaction> txns = {tx};
 
-BOOST_ASSERT (::ChainActive()
+    int nHeight = ::ChainActive().Height();
 
-.
+    // Mining a block with a asset create transaction
+    {
+        auto block = std::make_shared<CBlock>(CreateBlock(txns, coinbaseKey));
+        EnsureChainman(m_node).ProcessNewBlock(Params(), block, true, nullptr);
 
-Height()
-
-== nHeight + 2);
-BOOST_ASSERT(block
-->
+        BOOST_ASSERT(::ChainActive().Height() == nHeight + 1);
+        BOOST_ASSERT(block->GetHash() == ::ChainActive().Tip()->GetBlockHash());
+    }
 
-GetHash()
+    std::string assetId = tx.GetHash().ToString();
 
-== ::ChainActive().Tip()->GetBlockHash());
-}
+    tx = CreateMintAssetTx(*m_node.mempool, utxos, coinbaseKey, assetId);
 
-BOOST_ASSERT(passetsCache
-->
-GetAssetMetaData(assetId, asset
-));
+    {
+        auto block = std::make_shared<CBlock>(CreateBlock({tx}, coinbaseKey));
+        EnsureChainman(m_node).ProcessNewBlock(Params(), block, true, nullptr);
 
-BOOST_ASSERT(asset
-.circulatingSupply == 10 * COIN);
+        BOOST_ASSERT(::ChainActive().Height() == nHeight + 2);
+        BOOST_ASSERT(block->GetHash() == ::ChainActive().Tip()->GetBlockHash());
+    }
 
-{
-//mismach uniqueid
-CMutableTransaction tx2;
+    CAssetMetaData asset;
+    BOOST_ASSERT(passetsCache->GetAssetMetaData(assetId, asset));
 
-CKey key;
-key.MakeNewKey(false);
-CScript scriptPubKey = GetScriptForDestination(key.GetPubKey().GetID());
-CAssetTransfer assetTransfer(assetId, 1 * COIN, 8);
-assetTransfer.
-BuildAssetTransaction(scriptPubKey);
-CTxOut out(0, scriptPubKey);
-tx2.vout.
-push_back(out);
+    BOOST_ASSERT(asset.circulatingSupply == 100 * COIN);
 
-tx2.vin.
-push_back(CTxIn(COutPoint(tx.GetHash(), 0))
-);
+    {
```

Comment:
```suggestion
    // Allow TX index to catch up with the block index.
    g_txindex->BlockUntilSyncedToCurrentChain();
    
    {
```
here need to wait for the index too

Reply:
Thanks - fixed.